src/main/java/com/odysseusinc/arachne/portal/service/mail/ArachneMailSender.java

```diff
@@ -66,31 +67,33 @@ public ArachneMailSender(JavaMailSender mailSender) {
     }
 
     public void send(ArachneMailMessage mailMessage) {
-
-        try {
-            MimeMessage message = mailSender.createMimeMessage();
-            MimeMessageHelper helper;
-            helper = new MimeMessageHelper(message, true);
-            helper.setSubject(mailMessage.getSubject().replaceAll("\\$\\{app-title\\}", appTitle));
-            helper.setFrom(from, mailMessage.getFromPersonal().replaceAll("\\$\\{app-title\\}", appTitle));
-            helper.setTo(mailMessage.getUser().getEmail());
-            URL templateUrl = this.getClass().getResource( PATH_TO_TEMPLATES + mailMessage.getTemplate() + NAME + EXTENSION);
-            String htmlString = buildContent(mailMessage.getTemplate(), mailMessage.getParameters());
-            if (templateUrl != null) {
-                File textTemplate = new File(templateUrl.getPath());
-                if (!textTemplate.isDirectory()) {
-                    helper.setText(buildContent(mailMessage.getTemplate() + NAME, mailMessage.getParameters()), htmlString);
+        Runnable runnable = () -> {
```

Comment:
In case of regular registration, either:  remove timeout for the case when a user is not registered actually, or send synchronously